template.tpl

```diff
@@ -116,27 +101,23 @@ setCookie('_km', cid, { 'max-age': 30 * 24 * 60 * 60, 'secure': true });
 // If event is a seedtag.timer then it's a qualified visit, we can record a close.
 let isTimer = copyFromDataLayer('event') === 'seedtag.timer';
 
+// Optional eventData could be set in data layer
+let eventData = copyFromDataLayer('eventData');
+
 var sUrl = 'https://t.kmtx.io/s?' +
-    'aid=' + encodeUriComponent(data.aid) +
+    'pid=' + encodeUriComponent(data.pid) +
     '&cid=' + cid +
     '&eid=' + uuid() +
     '&a=' + (isTimer ? 'close' : data.type) +
-    '&v=gtm_1' +
+    '&ed=' + (eventData === undefined ? '' : encodeUriComponent(eventData)) +
+    '&v=gtm_2' +
     '&url=' + encodeUri(getUrl()) +
     '&ref=' + encodeUri(getReferrerUrl()) +
     '&ts=' + getTimestampMillis() +
     '&trk=trkid' +
     '&t=img';
 
-sendPixel(sUrl, data.gtmOnSuccess, data.gtmOnFailure);
-
-if (data.sid && !isTimer) {
-    var tsUrl = 'https://t.kmtx.io/ts?' +
-        'aid=' + encodeUriComponent(data.aid) +
-        '&sid=' + encodeUriComponent(data.sid);
-
-    sendPixel(tsUrl, data.gtmOnSuccess, data.gtmOnFailure);
-}
+sendPixel(sUrl, data.gtmOnSuccess(), data.gtmOnFailure());
```

Comment:
Nop ! you want to pass the function as argument, not calling the function.

```
sendPixel(sUrl, data.gtmOnSuccess, data.gtmOnFailure);
```

Reply:
ha yes my bad